packages/kokoas-server/src/handleRequest/reqDownloadEstimateForCustomer/conversions/convertEstimateForCustomerById.ts

```diff
@@ -0,0 +1,269 @@
+    fileName: '見積書',
+    fileType: 'xlsx',
+  });
+
+  // Read excel file.
+  let workbook = new excel.Workbook();
+  let pageNumber = 1;
+
+  workbook = await workbook.xlsx.readFile(templateFilePath);
+
+  const estDataId = formatDataId(dataId.value);
+
+  const initializeWorksheet = (sheetName: string) => {
+    const ws = workbook.getWorksheet(sheetName);
+    ws.getCell('H1').value = `Page.   ${pageNumber++}`;
+    ws.getCell('A1').value = `工事番号:${estDataId}`;
+    return ws;
+  };
+
+  /****************
+   * 見積書見出
+   ***************/
+  (function shFrontPage() {
+    const ws = workbook.getWorksheet('見積書見出');
+    // ページ情報
+    ws.getCell('AS4').value = `Page.   ${pageNumber++}`;
+
+    // 日付
+    ws.getCell('AS2').value = new Date().toLocaleString();
+
+    // 見積管理番号
+   
+    ws.getCell('AS3').value = `見積管理No.:${estDataId}`;
+
+    // 顧客名
+    ws.getCell('B5').value = `${customerName.value} 様`;
+
+    // 見積金額
+    ws.getCell('W13').value = totalAmountAfterTax;
+
+    // 工事費
+    ws.getCell('W16').value = totalAmountBeforeTax;
+
+    // 消費税
+    ws.getCell('AA17').value = totalTaxAmount;
+
+    // 工事番号
+    ws.getCell('H22').value = estDataId.slice(0, -3);
+
+    // 工事名
+    ws.getCell('H23').value = projName.value;
+
+    // 工事場所
+    ws.getCell('H24').value = addressBuilder({
+      postal: postal.value,
+      address1: address1.value,
+      address2: address2.value,
+    });
+  })();
+  
+
+  /****************
+   * 見積書内訳
+   * **************/
+  (function shSummary() {
+    const ws = initializeWorksheet('見積内訳');
+
+    // 非割引額
+    ws.getCell('G3').value = totalAmountBeforeDiscount;
+    ws.getCell('G5').value = totalAmountBeforeDiscount;
+
+
+    // 割引額
+    ws.getCell('G6').value = totalDiscountAmount;
+
+    // 税額
+    ws.getCell('G7').value = totalTaxAmount;
+
+    // 合計
+    ws.getCell('G8').value = totalAmountAfterTax;
+  })();
+
+
+  /************
+   * 内訳明細
+   ***********/
+
+  // 見積をグループ化
+  const groupedEstItems = groupEstItems(estimatesTable);
+  const maxRows = 19;
+  const rowOffset = 2; // ３行目から開始
+
+  (function shGroupedDetails() {
+    
+    let currRowIdx = 1;
+    let currSheetIdx = 1;
+    
+    let ws = initializeWorksheet(`内訳明細 (${currSheetIdx})`);
+    let rowIdx =  currRowIdx + rowOffset;
+
+    for ( const [key, value] of groupedEstItems) {
+      const  { groupAmountBeforeTax } = value;
+      if (groupAmountBeforeTax < 0) continue; // 割引額は除外
+
+      if (currRowIdx > maxRows) {
+      // 次のシートへ
+        currRowIdx = 1;
+        ws = initializeWorksheet(`内訳明細 (${++currSheetIdx})`);
+      }
+
+      rowIdx =  currRowIdx + rowOffset;
+      ws.getCell(`A${rowIdx}`).value = `${currRowIdx}. ${key}`;
+      ws.getCell(`D${rowIdx}`).value = '式';
+      ws.getCell(`E${rowIdx}`).value = 1;
+      ws.getCell(`G${rowIdx}`).value = groupAmountBeforeTax;
+
+      currRowIdx++; // 次の行へ
+
+    }
+
+    ws.getCell(`A${rowIdx}`).value = '《 合 計 》';
+    ws.getCell(`G${rowIdx}`).value = totalAmountBeforeDiscount;
```

Comment:
```suggestion
    rowIdx += 1;

    ws.getCell(`A${rowIdx}`).value = '《 合 計 》';
    ws.getCell(`G${rowIdx}`).value = totalAmountBeforeDiscount;
```

rowIdxの更新忘れにより、1項目分を上書きしてしまっているようです。

Reply:
気づきませんでした。ありがとうございます！ eab93b2f で対応しました。